compiler/ballerina-lang/src/main/java/io/ballerina/projects/directory/BuildProject.java

```diff
@@ -138,15 +138,31 @@ private Optional<Path> modulePath(ModuleId moduleId) {
         return Optional.empty();
     }
 
+    private Optional<Path> generatedModulePath(ModuleId moduleId) {
+        if (currentPackage().moduleIds().contains(moduleId)) {
+            if (currentPackage().getDefaultModule().moduleId() == moduleId) {
+                return Optional.of(sourceRoot);
+            } else {
+                return Optional.of(sourceRoot.resolve(ProjectConstants.GENERATED_MODULES_ROOT).resolve(
+                        currentPackage().module(moduleId).moduleName().moduleNamePart()));
+            }
+        }
+        return Optional.empty();
+    }
+
     @Override
     public Optional<Path> documentPath(DocumentId documentId) {
         for (ModuleId moduleId : currentPackage().moduleIds()) {
             Module module = currentPackage().module(moduleId);
             Optional<Path> modulePath = modulePath(moduleId);
+            Optional<Path> generatedModulePath = generatedModulePath(moduleId);
             if (module.documentIds().contains(documentId)) {
-                if (modulePath.isPresent()) {
-                    return Optional.of(modulePath.get().resolve(module.document(documentId).name()));
+                if (generatedModulePath.isPresent()) {
+                    if (Files.exists(generatedModulePath.get().resolve(module.document(documentId).name()))) {
+                        return Optional.of(generatedModulePath.get().resolve(module.document(documentId).name()));
+                    }
                 }
+                return Optional.of(modulePath.get().resolve(module.document(documentId).name()));
```

Comment:
Shouldn't we keep the check at L147?

Reply:
Shall we keep the existing checks as it and add another else-if for generated directory after L172?